### PR TITLE
[codex] Align root SFT samples with standardized records

### PR DESCRIPTION
**Pull request by neubig**
Part of #218.

## Summary

- Add an unskipped validation test requiring each root `sample_sft.json` to preserve the corresponding `sample_std.json` record count, order, and IDs.
- Regenerate root OpenHands SFT samples for `androidcontrol`, `eto`, `omniact`, `turkingbench`, `webarena_successful`, and `weblinx` so they align with their standardized samples.
- Preserve WebLINX unresolved XPath actions during OpenHands web SFT conversion by falling back to the original XPath when BrowserGym BID lookup fails.
- Move AXTree BID lookup diagnostics out of dataset directories and into ignored `temp/` scratch files so conversion does not create forbidden top-level dataset JSON.

## Notes

This intentionally scopes the fix to standardized-to-root-SFT alignment. Raw-to-standardized count differences remain for separate follow-up work because some raw records expand or filter during standardization.

## Validation

- `uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/test_std_to_sft_conversion.py::test_sample_std_and_root_sft_records_align tests/test_openhands_sft_role_preservation.py -q`
- `uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/test_std_to_sft_conversion.py tests/test_openhands_sft_role_preservation.py -q`
- `uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/ -q`
- `uv run --python 3.10 --isolated --with ruff ruff check agents/openhands/std_to_sft.py scripts/html_to_axtree.py tests/test_openhands_sft_role_preservation.py tests/test_std_to_sft_conversion.py`
- `git diff --check`

## Evidence

Local validation output after resolving the merge conflict:

```text
$ uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/test_std_to_sft_conversion.py::test_sample_std_and_root_sft_records_align tests/test_openhands_sft_role_preservation.py -q
55 passed in 0.28s

$ uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/test_std_to_sft_conversion.py tests/test_openhands_sft_role_preservation.py -q
93 passed, 12 skipped in 0.37s

$ uv run --python 3.10 --isolated --with-requirements requirements.txt python -m pytest tests/ -q
425 passed, 14 skipped, 4 warnings in 17.92s

$ uv run --python 3.10 --isolated --with ruff ruff check agents/openhands/std_to_sft.py scripts/html_to_axtree.py tests/test_openhands_sft_role_preservation.py tests/test_std_to_sft_conversion.py
All checks passed!

$ git diff --check
# no output
```

End-to-end sample regeneration evidence for one aligned root SFT dataset (`androidcontrol`), running `sample_raw.json -> raw_to_standardized.py -> agents/openhands/std_to_sft.py` into ignored `temp/evidence/` files:

```text
$ export MY_DATASET=androidcontrol PYTHONPATH=$PWD:$PYTHONPATH
$ cat datasets/androidcontrol/sample_raw.json | python scripts/json_to_jsonl.py | python datasets/androidcontrol/raw_to_standardized.py | python scripts/jsonl_to_json.py > temp/evidence/androidcontrol_sample_std.json
$ cat temp/evidence/androidcontrol_sample_std.json | python scripts/json_to_jsonl.py | python agents/openhands/std_to_sft.py --is_web=no --api_env=execute_bash | python scripts/jsonl_to_json.py > temp/evidence/androidcontrol_sample_sft.json
$ python - <<'PY'
import json
std=json.load(open("temp/evidence/androidcontrol_sample_std.json"))
sft=json.load(open("temp/evidence/androidcontrol_sample_sft.json"))
committed_sft=json.load(open("datasets/androidcontrol/sample_sft.json"))
std_ids=[r["id"] for r in std]
sft_ids=[r["id"] for r in sft]
print("generated std records:", len(std), "ids:", std_ids)
print("generated sft records:", len(sft), "ids:", sft_ids)
print("committed root sft matches generated:", sft == committed_sft)
print("std/sft ids aligned:", std_ids == sft_ids)
print("first SFT turns:", len(sft[0]["conversations"]))
PY
generated std records: 3 ids: ['0', '20', '40']
generated sft records: 3 ids: ['0', '20', '40']
committed root sft matches generated: True
std/sft ids aligned: True
first SFT turns: 8
```

_This PR description update was generated by an AI agent (OpenHands) on behalf of the user._

### CODE REVIEW

**Comment by neubig** [comment]
@openhands fix merge conflicts and /iterate

File: tests/test_std_to_sft_conversion.py [hunk 1]
```diff
@@ -32,6 +32,34 @@ def get_subdirectories(directory):
...
+        sft_data = json.load(f)
+
+    assert len(std_data) == len(sft_data), (
+        f"Number of samples in std ({len(std_data)}) and root sft ({len(sft_data)}) "
+        f"don't match in {subdir}"
+    )
+
+    std_ids = [sample["id"] for sample in std_data]
+    sft_ids = [sample["id"] for sample in sft_data]
+    assert std_ids == sft_ids, f"Sample ids don't match in {subdir}: {std_ids} vs {sft_ids}"
```
[the remark this answers is not in the document]
Comment (neubig) [reply tests/test_std_to_sft_conversion.py:62]: Addressed in 9db4e7b by adding a note that this test is scoped to stage alignment only and that duplicate source IDs remain tracked under #218.

_This reply was generated by an AI agent (OpenHands) on behalf of the user._

File: scripts/html_to_axtree.py [hunk 2]
```diff
@@ -63,9 +63,8 @@ def get_bid(self, id, x_path: str, chunk) -> str:
                     "raw_html": self.last_html,
                 }
             )
-            with open(
-                f"./datasets/{self.dataset}/{self.dataset}_{chunk}_bid_errors.json", "w"
-            ) as f:
+            os.makedirs("temp", exist_ok=True)
```
[the remark this answers is not in the document]
Comment (neubig) [reply scripts/html_to_axtree.py]: Done in 43f0848: AXTree scratch HTML files and BID diagnostics now resolve under the repository-root temp directory regardless of the current working directory.

_This reply was generated by an AI agent (OpenHands) on behalf of the user._
